frontend/apps/filemanager/filemanagermenu.lua

```diff
@@ -790,6 +800,74 @@ dbg:guard(FileManagerMenu, 'setUpdateItemTable',
         end
     end)
 
+function FileManagerMenu:moveBookMetadata()
+    local DocSettings = require("docsettings")
+    local FileChooser = self.ui.file_chooser
+    local function scanPath()
+        local sys_folders = { -- do not scan sys_folders
+            ["/dev"]  = true,
+            ["/proc"] = true,
+            ["/sys"]  = true,
+        }
+        local books_to_move = {}
+        local dirs = {FileChooser.path}
+        while #dirs ~= 0 do
+            local new_dirs = {}
+            for _, d in ipairs(dirs) do
+                local ok, iter, dir_obj = pcall(lfs.dir, d)
+                if ok then
+                    for f in iter, dir_obj do
+                        local fullpath = "/" .. f
+                        if d ~= "/" then
+                            fullpath = d .. fullpath
+                        end
+                        local attributes = lfs.attributes(fullpath) or {}
+                        if attributes.mode == "directory" and f ~= "." and f ~= ".."
+                                and FileChooser:show_dir(f) and not sys_folders[fullpath] then
+                            table.insert(new_dirs, fullpath)
+                        elseif attributes.mode == "file" and not util.stringStartsWith(f, "._")
+                                and FileChooser:show_file(f) and DocSettings:hasSidecarFile(fullpath)
+                                and lfs.attributes(DocSettings:getSidecarFile(fullpath), "mode") ~= "file" then
+                            table.insert(books_to_move, fullpath)
+                        end
+                    end
+                end
+            end
+            dirs = new_dirs
+        end
+        return books_to_move
+    end
+    UIManager:show(ConfirmBox:new{
+        text = _("Scan current folder and subfolders for book metadata?"),
```

Comment:
One could think "Having set my metadata location to koreader/docsettings/, I won't find any if I scan from my library root directory".
Probably more explicite with:
`Scan books in current folder and subfolders for their metadata location?`

Reply:
This suggestion sounds good to me.